### PR TITLE
gh-92240 : Include release dates for "What's New In Python 3.X"

### PR DESCRIPTION
**Pull request by georgically**
Added release dates for
"What's New in Python 3.X" for 3.0, 3.1, 3.2, 3.8 and 3.10

#92240

### CODE REVIEW

File: Doc/whatsnew/3.0.rst [hunk 1]
```diff
@@ -53,9 +53,9 @@
 
 This article explains the new features in Python 3.0, compared to 2.6.
 Python 3.0, also known as "Python 3000" or "Py3K", is the first ever
-*intentionally backwards incompatible* Python release.  There are more
-changes than in a typical release, and more that are important for all
-Python users.  Nevertheless, after digesting the changes, you'll find
+*intentionally backwards incompatible* Python release. Python 3.0 was released on 3 Dec 2008.
```
Comment (slateny) [inline Doc/whatsnew/3.0.rst]: I think that the date formatting should be (somewhat) consistent, at least for the whatsnew pages changed in this PR. Looking at [3.9's whatsnew](https://docs.python.org/3/whatsnew/3.9.html), it has it as 'October 5th, 2020', and 3.7 has it as 'June 27, 2018', so `month, day, year` seems like a good enough standard to go off in this case.

Comment (georgically) [reply Doc/whatsnew/3.0.rst]: I agree with date formatting consistency. Made adjustments to all versions.

File: Doc/whatsnew/2.1.rst [hunk 2]
```diff
@@ -27,7 +27,7 @@ of new releases, with a new release coming every 6 to 9 months. 2.1 is the first
 release to come out at this faster pace, with the first alpha appearing in
 January, 3 months after the final version of 2.0 was released.
 
-The final release of Python 2.1 was made on April 17, 2001.
+The final release of Python 2.1 was released on April 17, 2001.
```
Comment (AA-Turner) [inline Doc/whatsnew/2.1.rst]: "release ... was released" doesn't scan terribly well. I don't think this line needs to change.

A

**Comment by AA-Turner** [comment]
Please also update NEWS as you've expanded to 2.X.

A

File: Doc/whatsnew/2.1.rst (hunk 2 above)
Comment (georgically) [reply Doc/whatsnew/2.1.rst]: This change was made to retain consistency throughout the entire documentation among different versions. TBH, it is definitely not made on April 17, 2001 rather released.

Comment (AA-Turner) [reply Doc/whatsnew/2.1.rst]: Changing the text of a 21-year old whatsnew document purely for consistency seems iffy, I remain -1 on it.

A

File: Doc/whatsnew/2.3.rst [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 .. $Id: whatsnew23.tex 54631 2007-03-31 11:58:36Z georg.brandl $
 
-This article explains the new features in Python 2.3.  Python 2.3 was released
+This article explains the new features in Python 2.3. Python 2.3 was released
```
Comment (AA-Turner) [inline Doc/whatsnew/2.3.rst]: Similarly, double spacing was commonly used in documents in the past (and still is by some authors) -- see the remainder of What's New in Python 2.3 where `.  ` (full stop with two literal space characters) is used frequently. Please revert this change.

A

Comment (AA-Turner) [reply Doc/whatsnew/2.3.rst]: & also please revert where you've changed a double space to a single space in other documents (e.g. 3.0).

File: Doc/whatsnew/3.0.rst [hunk 4]
```diff
@@ -53,9 +53,9 @@
 
 This article explains the new features in Python 3.0, compared to 2.6.
 Python 3.0, also known as "Python 3000" or "Py3K", is the first ever
-*intentionally backwards incompatible* Python release.  There are more
-changes than in a typical release, and more that are important for all
-Python users.  Nevertheless, after digesting the changes, you'll find
+*intentionally backwards incompatible* Python release. Python 3.0 was released on Dec 3, 2008.
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.0.rst]:
```suggestion
*intentionally backwards incompatible* Python release. Python 3.0 was released on December 3, 2008.
```

File: Doc/whatsnew/2.1.rst (hunk 2 above)
Comment (JelleZijlstra) [reply Doc/whatsnew/2.1.rst]: Agree, this change is unnecessary

File: Doc/whatsnew/3.10.rst [hunk 5]
```diff
@@ -46,8 +46,7 @@
    This saves the maintainer the effort of going through the git log
    when researching a change.
 
-This article explains the new features in Python 3.10, compared to 3.9.
-
+This article explains the new features in Python 3.10, compared to 3.9. Python 3.10 was released on Oct 4, 2021.
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.10.rst]:
```suggestion
This article explains the new features in Python 3.10, compared to 3.9.
Python 3.10 was released on October 4, 2021.
```

File: Doc/whatsnew/3.2.rst [hunk 6]
```diff
@@ -48,7 +48,7 @@
    This saves the maintainer the effort of going through the SVN log
    when researching a change.
 
-This article explains the new features in Python 3.2 as compared to 3.1.  It
+This article explains the new features in Python 3.2 as compared to 3.1. Python 3.2 was released on 20 Feb, 2011. It
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.2.rst]:
```suggestion
This article explains the new features in Python 3.2 as compared to 3.1.
Python 3.2 was released on February 20, 2011. It
```

File: Doc/whatsnew/3.8.rst [hunk 7]
```diff
@@ -44,7 +44,7 @@
 
 :Editor: Raymond Hettinger
 
-This article explains the new features in Python 3.8, compared to 3.7.
+This article explains the new features in Python 3.8, compared to 3.7. Python 3.8 was released on Oct 14, 2019.
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.8.rst]:
```suggestion
This article explains the new features in Python 3.8, compared to 3.7.
Python 3.8 was released on October 14, 2019.
```

File: Misc/NEWS.d/next/Documentation/2022-05-18-23-58-26.gh-issue-92240.bHvYiz.rst [hunk 8]
```diff
@@ -0,0 +1,6 @@
+Added release dates for
+"What's New in Python 3.X" for 3.0, 3.1, 3.2, 3.8 and 3.10
+2.1 - changed 'made' to 'released' to retain consistency.
+2.3, 2.6, 3.2 - whitespace and punctuation adjusted.
+2.5 - release date added.
+3.9 - date format adjusted.
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Documentation/2022-05-18-23-58-26.gh-issue-92240.bHvYiz.rst]:
```suggestion
```

This is unnecessary detail. Arguably the news entry isn't needed at all.

**Review by JelleZijlstra — COMMENTED** [review]
Thanks for your contribution!

File: Doc/whatsnew/2.3.rst (hunk 3 above)
Comment (georgically) [reply Doc/whatsnew/2.3.rst]: Reverted mentioned double space. 3.0 release date was added later because it was missing. But it's adjusted to double spacing now.

File: Doc/whatsnew/3.1.rst [hunk 9]
```diff
@@ -46,7 +46,7 @@
    This saves the maintainer the effort of going through the SVN log
    when researching a change.
 
-This article explains the new features in Python 3.1, compared to 3.0.
+This article explains the new features in Python 3.1, compared to 3.0. Python 3.1 was released on June 27, 2009.
```
Comment (AA-Turner) [inline Doc/whatsnew/3.1.rst]:
```suggestion
This article explains the new features in Python 3.1, compared to 3.0.
Python 3.1 was released on June 27, 2009.
```

**Review by JelleZijlstra — APPROVED** [review]
